chpl/chpl-api/src/test/java/gov/healthit/chpl/util/DeprecatedFieldExplorerTest.java

```diff
@@ -0,0 +1,869 @@
+        assertTrue(deprecatedFieldNames.contains("pendingSurveillance" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformityCloseDate"));
+    }
+
+    @Test
+    public void findDeprecatedFields_Surveillance() {
+        Set<String> deprecatedFieldNames = deprecatedFieldExplorer.getDeprecatedFieldsForClass(Surveillance.class);
+
+        assertNotNull(deprecatedFieldNames);
+        assertEquals(9, deprecatedFieldNames.size());
+        assertTrue(deprecatedFieldNames.contains("startDate"));
+        assertTrue(deprecatedFieldNames.contains("endDate"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "status"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "dateOfDetermination"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "capApprovalDate"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "capStartDate"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "capEndDate"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "capMustCompleteDate"));
+        assertTrue(deprecatedFieldNames.contains("requirements" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformities" + DeprecatedFieldExplorer.FIELD_SEPARATOR + "nonconformityCloseDate"));
+    }
+
+    @Test
+    public void findDeprecatedFields_ApiKey() {
+        Set<String> deprecatedFieldNames = deprecatedFieldExplorer.getDeprecatedFieldsForClass(ApiKey.class);
+
+        assertNotNull(deprecatedFieldNames);
+        assertEquals(1, deprecatedFieldNames.size());
+        assertTrue(deprecatedFieldNames.contains("apiKey"));
+    }
+
+    @Test
+    public void findDeprecatedFields_CertificationIdResults() {
+        Set<String> deprecatedFieldNames = deprecatedFieldExplorer.getDeprecatedFieldsForClass(CertificationIdResults.class);
+
+        assertNotNull(deprecatedFieldNames);
+        assertEquals(1, deprecatedFieldNames.size());
+        assertTrue(deprecatedFieldNames.contains("isValid"));
```

Comment:
I like having both the count and the list of expected ones. I feel like these tests will help find any unexpectedly deprecated elements

And I just looked to see if there were any that were expecting 0, and found some, so now the only thing that we would miss should be an entirely new response Class that isn't in this test